tests/ui/consts/const-eval/raw-bytes.rs

```diff
@@ -1,6 +1,7 @@
 // stderr-per-bitwidth
 // ignore-endian-big
 // ignore-tidy-linelength
+// ignore-debug debug assertions catch some UB too early
```

Comment:
This is bad. It is fairly common that developers have debug assertions enabled locally so this now means they can't run or bless this test any more.

Reply:
The diff seems to be caused by `Alignment::new_unchecked`. IMO we should either un-do the PR for that or use a different way of creating invalid layouts for this test (that avoids the assertion).

Reply:
Follow-up: https://github.com/rust-lang/rust/pull/118358